src/main/features/claim/guards/newFeaturesConsentGuard.ts

```diff
@@ -0,0 +1,19 @@
+import * as express from 'express'
+import { ClaimStoreClient } from 'claims/claimStoreClient'
+import { ForbiddenError } from 'errors'
+
+const claimStoreClient = new ClaimStoreClient()
+
+export class NewFeaturesConsentGuard {
+
+  static requestHandler (req: express.Request, res: express.Response, next: express.NextFunction) {
+    claimStoreClient.retrieveUserRoles(res.locals.user)
+      .then(roles => {
+        if (roles.length !== 0 && roles.some(role => role.includes('cmc-new-features-consent'))) {
+          return res.render(new ForbiddenError().associatedView)
```

Comment:
We should return 404 in that case, not forbidden. Also we should an Error instead of rendering page manually.

Reply:
I've done a redirect to the task list instead, I couldn't get throwing an error to work from the async guard factory it just hung